### PR TITLE
Check that tests compile during build

### LINKED ISSUE #169: Add check for tests compilation
This would prevent compilation errors in the test code from going unnoticed.

This can be achieved by adding another typescript config json file for the test source and using the --noEmit flag on `tsc` 

Example of doing this in zowe CLI: https://github.com/zowe/zowe-cli/blob/master/package.json#L44

### PR DESCRIPTION
**Pull request by ChrisBoehmCA**
Fix existing compilation errors
Fix #169

### CODE REVIEW

File: __tests__/src/packages/profiles/src/constants/BasicProfileManagerTestContants.ts [hunk 1]
```diff
@@ -62,15 +62,15 @@ const secureOrangeProfile: ICommandProfileTypeConfiguration = {
       username: {
         optionDefinition: {
           description: "The username of the secure_orange",
-          type: "username",
+          type: "string",
```
Comment (zFernand0) [inline __tests__/src/packages/profiles/src/constants/BasicProfileManagerTestContants.ts]: Thanks for updating

File: package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
     "build:packages": "gulp build",
     "build": "gulp build",
     "postbuild": "gulp build:install-all-cli-dependencies && gulp build:all-clis",
+    "checkTestsCompile": "echo \"Checking that test source compiles\" && tsc --noEmit -p tsconfig-tests.json",
```
Comment (zFernand0) [inline package.json]: What do you think about adding it in the postbuild npm script?

File: Jenkinsfile [hunk 3]
```diff
@@ -72,7 +72,14 @@ node('ca-jenkins-agent') {
 
     // Build the application
     pipeline.build(
-        timeout: [time: 5, unit: 'MINUTES']
+        operation: {
+            sh "npm run build"
+            sh "npm run checkTestsCompile"
```
Comment (zFernand0) [inline Jenkinsfile]: If we decide to make `checkTestCOmpile` part of the `postbuild` npm script, then we could remove this `operation`

File: package.json (hunk 2 above)
Comment (ChrisBoehmCA) [reply package.json]: That's fine with me. I will change it

**Comment by zFernand0** [comment]
merging

**Pull request merged**